app/src/pages/DustyEthLockPage.tsx

```diff
@@ -0,0 +1,215 @@
+            isLoading: true,
+            networkType: '',
+            isProcessing: false,
+            allLockEvents: [],
+            error: null,
+            fetchingLockData: true,
+        };
+    }
+
+    // used for fetching data periodically
+    timerInterval: any;
+
+    // get and set the web3 state when the component is mounted
+    componentDidMount = async () => {
+        const web3State = await connectWeb3(LockSeason.First);
+        this.setState(web3State);
+
+        // checks if account has changed in MetaMask
+        if ((window as any).ethereum.on) {
+            (window as any).ethereum.on('accountsChanged', this.handleAccountChange);
+        }
+
+        this.setState({ networkType: await this.state.web3.eth.net.getNetworkType() });
+
+        this.timerInterval = setInterval(() => {
+            this.getLockData().then(() => {
+                this.setState({ isLoading: false });
+            });
+        }, 1000);
+    };
+
+    isMainnet = () => {
+        return this.state.networkType === 'main';
+    };
+
+    componentWillUnmount = () => {
+        clearInterval(this.timerInterval);
+        removeWeb3Event();
+    };
+
+    // called when the user changes MetaMask account
+    handleAccountChange = () => {
+        // refresh the page
+        window.location.reload(false);
+    };
+
+    getLockData = async () => {
+        try {
+            // get all the lock events from the chain
+            const allLocks = await getAllLockEvents(this.state.web3, this.state.contract);
+
+            this.setState({ allLockEvents: allLocks });
+        } catch (error) {
+            this.setState({ error });
+        }
+    };
+
+    handleSubmit = async (formInputVal: LockInput) => {
+        // checks user input
+        if (formInputVal.amount > new BN(0) && formInputVal.duration) {
+            this.setState({ isProcessing: true });
+            //console.log(formInputVal);
+            // return a default address if user input is empty
+            const introducer = defaultAffiliation(formInputVal.affiliation).toLowerCase();
+            const { accounts, contract } = this.state;
+            try {
+                // check user input
+                if (introducer === accounts[0]) {
+                    toast.error('You cannot affiliate yourself');
+                } else if (introducer && !Web3.utils.isAddress(introducer)) {
+                    toast.error('Please input a valid Ethereum address');
+                } else if (!isRegisteredEthAddress(introducer)) {
+                    toast.error('The given introducer is not registered in the affiliation program!');
+                } else {
+                    // convert user input to Wei
+                    const amountToSend = Web3.utils.toWei(formInputVal.amount, 'ether');
+
+                    // communicate with the smart contract
+                    await contract.methods.lock(formInputVal.duration, introducer).send({
```

Comment:
Real-time lock drop requires that transactions be sent to the Plasm side in addition to this, and There is. Ask Akru about this process.
https://docs.plasmnet.io/workshop-and- tutorial/real-time-lockdrop

Also, this process is the same for 2nd, 3rd and dusty Ethereum Lockdrop process. Therefore, it is better to make it will be functioned.

Reply:
This addition is a big change and should be your next PR.

Reply:
to summarize the real-time lockdrop UI:
there should be a lock section and a claim section
Submit lock tx UI is the same

Claim lock UI section should display:

```
{
  type: number,
  transactionHash: string,
  publicKey: string,
  duration: number,
  value: BigNumber(string)
}
```
users will submit this transaction directly to the plasmnet-app

Lockdrop UI will show the claim response `Storage::plasmLockdrop.claims` from the blockchain
is this correct?

Also, does that mean I should first merge this PR and open another one? If we want better UX with the ability to submit transactions directly from the lockdrop UI to plasm network, we'll need to find a safe way to import plasm private keys from one UI to another.
What's your thought @akru ?